service/src/main/java/uk/nhs/adaptors/gp2gp/transformJsonToXmlTool/TransformJsonToXml.java

```diff
@@ -0,0 +1,250 @@
+                String jsonString = jsonFileInputs.get(i);
+                String xmlResult = mapJsonToXml(jsonString);
+                String fileName = jsonFileNames.get(i);
+                writeToFile(xmlResult, fileName);
+            }
+        } catch (NHSNumberNotFound | UnreadableJsonFileException | NoJsonFileFound e) {
+            LOGGER.error("error: " + e.getMessage());
+        }
+        LOGGER.error("end");
+    }
+
+    private static InputWrapper getFiles() throws UnreadableJsonFileException, NoJsonFileFound {
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        if (files == null || files.length == 0) {
+            throw new NoJsonFileFound("No json files found");
+        }
+
+        LOGGER.info("Processing " + files.length + " files from location: " + JSON_FILE_INPUT_PATH);
+
+        Arrays.stream(files)
+            .peek(file -> LOGGER.info("Parsing file: {}", file.getName()))
+            .filter(file -> FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("json"))
+            .forEach(file -> {
+                String jsonAsString = null;
+                try {
+                    jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + file.getName());
+                } catch (Exception e) {
+                    throw new UnreadableJsonFileException("Cant read file " + file.getName());
+                }
+                jsonStringInputs.add(jsonAsString);
+                fileNames.add(file.getName());
+            });
+        return InputWrapper.builder().jsonFileInputs(jsonStringInputs).jsonFileNames(fileNames).build();
+    }
+
+    private static String mapJsonToXml(String jsonAsStringInput) throws NHSNumberNotFound {
+        final Bundle bundle = new FhirParseService().parseResource(jsonAsStringInput, Bundle.class);
+
+        MESSAGE_CONTEXT.initialize(bundle);
+
+        GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition;
+
+        getGpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
+            .nhsNumber(extractNhsNumber(jsonAsStringInput))
+            .conversationId("6910A49D-1F97-4AA0-9C69-197EE9464C76")
+            .requestId("17A3A644-A4EB-4C0A-A870-152D310FD1F8")
+            .fromOdsCode("GP2GPTEST")
+            .toOdsCode("GP2GPTEST")
+            .toAsid("GP2GPTEST")
+            .fromAsid("GP2GPTEST")
+            .build();
+
+        final EhrExtractTemplateParameters ehrExtractTemplateParameters =
+            EHR_EXTRACT_MAPPER.mapBundleToEhrFhirExtractParams(getGpcStructuredTaskDefinition, bundle);
+
+        final String ehrExtractContent = EHR_EXTRACT_MAPPER.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        final String hl7TranslatedResponse = OUTPUT_MESSAGE_WRAPPER_MAPPER.map(getGpcStructuredTaskDefinition, ehrExtractContent);
+
+        MESSAGE_CONTEXT.resetMessageContext();
```

Comment:
Would be better to wrap code above into a `try-catch` and put the `MESSAGE_CONTEXT.resetMessageContext()` in the `finally` block